vars/githubPr.groovy

```diff
@@ -97,18 +99,66 @@ def getHistoryText(builds) {
   return "### History\n${list}"
 }
 
+def getTestFailuresMessage() {
+  def failures = testUtils.getFailures()
+  if (!failures) {
+    return ""
+  }
+
+  def messages = []
+
+  failures.take(5).each { failure ->
+    messages << """
+---
+
+### [Test Failures](${env.BUILD_URL}testReport)
+<details><summary>${failure.fullDisplayName}</summary>
+
+[Link to Jenkins](${failure.url})
+
+```
+${failure.stdOut}
+```
+</details>
+
+---
+    """
+  }
+
+  if (failures.size() > 3) {
+    messages << "and ${failures.size() - 3} more failures, only showing the first 3."
+  }
+
+  return messages.join("\n")
+}
+
 def getNextCommentMessage(previousCommentInfo = [:]) {
-  info = previousCommentInfo ?: [:]
+  def info = previousCommentInfo ?: [:]
   info.builds = previousCommentInfo.builds ?: []
 
   def messages = []
+  def status = buildUtils.getBuildStatus()
 
-  if (buildUtils.getBuildStatus() == 'SUCCESS') {
+  if (status == 'SUCCESS') {
     messages << """
       ## :green_heart: Build Succeeded
       * [continuous-integration/kibana-ci/pull-request](${env.BUILD_URL})
       * Commit: ${getCommitHash()}
     """
+  } else if(status == 'UNSTABLE') {
+    def message = """
```

Comment:
I considered moving to some kind of templating because of how this string handling is getting, but it still seems like probably more work than it's worth.

The whitespace is really particular to get the formatting of the comment right in a few places, which is why some of this is weird